Adding blog post about ZKHack awards (#162)
risc0/website: blog/2023-05-04-zkawards/index.md

---
slug: zkhack
title: Project Awards ZKHack 
authors: [kay]
tags: [zkhack, zksummit, lisbon]
---
# ZKHack Lisbon

We recently participated in [ZKHack Lisbon](https://www.zklisbon.com/), [ZK Hack](https://zkhack.dev/)'s first in-person hackathon.
Thanks to [everyone who contributed projects](https://zk-lisbon.devfolio.co/projects)!

> Below is a list of all [RISC Zero prize](https://zk-lisbon.devfolio.co/prizes?partner=RISC+Zero) winners from ZK Hack Lisbon. We've provided longer descriptions for these projects as an added thank you to the four teams that won $2000. 

Reviewer: Unclear reference with "these", I've suggested a possible reorder to clear this up.

```suggestion
Below is a list of all [RISC Zero prize](https://zk-lisbon.devfolio.co/prizes?partner=RISC+Zero) winners from ZK Hack Lisbon.
For the four teams that won $2000, we've also provided a longer description as an added thank you.
```

To support future hackathon participants, we've included *why* our judging team loved the top projects. Our judging criteria will vary from hackathon to hackathon, but we're excited to see development in emerging ZK application spaces, inventive uses of ZK tech, and projects that demonstrate the simplicity and power of ZK. And because doing is teaching, we love submissions that either contain self-explanatory educational content or insightful supplementary docs.

## RISC Zero Prize Categories

We encouraged participants at ZKHack Lisbon to submit two types of project entries:

| | Problem description | 
|---|---|
|Challenge #1 | Build an application that uses the RISC Zero zkVM. |
|Challenge #2| Write code that utilizes data serialization or deserialization crates in the guest. |

# RISC Zero Hackathon Prize Winners

## Winning Team Projects: $2000 Level

> ### Mindf*ck: Code Obfuscation For Fun And Profit

Reviewer: I see the submission does not censor the swearing -- I'm not sure we should either, I don't particularly see any need to keep a "clean" image, and am therefore a bit worried we're coming across as stodgy.

Pretty minor point though, I could be persuaded otherwise.

* Team: Zeroskill
* Category: Serialization and Deserialization (Challenge #2)
* [Project submission](https://devfolio.co/projects/mindfuck-0f98)
* [Project link](https://github.com/hananbeer/mindfuck)

> To obfuscate code execution, the Zeroskill team has added a Rust-based [Brainf\*ck](https://esolangs.org/wiki/Brainfuck) interpreter to the guest program. With this addition, the execution of an interpreted program is captured by the guest zkVM.

Reviewer: Hmm, can we adopt a slightly more playful tone here? I don't think we need a full tone rewrite, but also writing in Brainfuck is very loudly proclaiming "what we're doing is humorous!" and I'm worried that if there's nothing comedic in this section we might look like we're not in on the joke.

I've made a suggestion with a touch of humor, but I'm not sure whether it's actually good...
```suggestion
To obfuscate code execution (and to entertain judges and onlookers), the Zeroskill team has added a Rust-based [Brainf\*ck](https://esolangs.org/wiki/Brainfuck) interpreter to the guest program. With this addition, the execution of an interpreted program is captured by the guest zkVM.
```
